app/questionnaire/navigation.py

```diff
@@ -0,0 +1,151 @@
+import copy
+import logging
+
+from collections import defaultdict
+
+from app.data_model.answer_store import AnswerStore
+
+from app.helpers.schema_helper import SchemaHelper
+from app.questionnaire.location import Location
+from app.questionnaire.rules import evaluate_repeat
+
+logger = logging.getLogger(__name__)
+
+
+class Navigation(object):
+    """
+    Reads navigation config from the schema and returns collections of dicts that describe
+    completion status of each group
+    """
+    def __init__(self, survey_json, metadata=None, answer_store=None, completed_blocks=None):
```

Comment:
Originally it was, but I wanted to preserve the argument order that Navigator (now PathFinder) has and weirdly metadata comes before it there. Given metadata is optional and a kwarg can't precede an arg, hence it's ended up this way.

Additionally, originally the tests were modifying the content of the store without ever having created one, so this was a solution for that. I've already refactored the tests, so may well be able to resolve this now.